packages/core/src/validate/overrides.ts

```diff
@@ -5,67 +5,68 @@ import { ValidationError } from './run';
 export interface ValidationOptions {
   unsafeAllowCustomTypes?: boolean;
   unsafeAllowLinkedLibraries?: boolean;
+  unsafeAllow?: ValidationError['kind'][];
 }
 
-export function withValidationDefaults(opts: ValidationOptions): Required<ValidationOptions> {
-  return {
-    unsafeAllowCustomTypes: opts.unsafeAllowCustomTypes ?? false,
-    unsafeAllowLinkedLibraries: opts.unsafeAllowLinkedLibraries ?? false,
-  };
-}
+export const ValidationErrorUnsafeMessages: Record<ValidationError['kind'], string | undefined> = {
+  'state-variable-assignment': `    You are using the \`unsafeAllow.state-variable-assignment\` flag.\n`,
+  'state-variable-immutable': `    You are using the \`unsafeAllow.state-variable-immutable\` flag.\n`,
+  'external-library-linking':
+    `    You are using the \`unsafeAllow.external-library-linking\` flag to include external libraries.\n` +
+    `    Make sure you have manually checked that the linked libraries are upgrade safe.\n`,
+  'struct-definition':
+    `    You are using the \`unsafeAllow.struct-definition\` flag to skip storage checks for structs.\n` +
+    `    Make sure you have manually checked the storage layout for incompatibilities.\n`,
+  'enum-definition':
+    `    You are using the \`unsafeAllow.enum-definition\` flag to skip storage checks for enums.\n` +
+    `    Make sure you have manually checked the storage layout for incompatibilities.\n`,
+  constructor: `    You are using the \`unsafeAllow.constructor\` flag.\n`,
+  delegatecall: `    You are using the \`unsafeAllow.delegatecall\` flag.\n`,
+  selfdestruct: `    You are using the \`unsafeAllow.selfdestruct\` flag.\n`,
+  'inline-assembly': `    You are using the \`unsafeAllow.inline-assembly\` flag.\n`,
+};
 
-export function processExceptions(
-  contractName: string,
-  errorsToProcess: ValidationError[],
-  opts: ValidationOptions,
-): ValidationError[] {
-  const { unsafeAllowCustomTypes, unsafeAllowLinkedLibraries } = withValidationDefaults(opts);
-  let errors: ValidationError[] = errorsToProcess;
+export function withValidationDefaults(opts: ValidationOptions): Required<ValidationOptions> {
+  const unsafeAllowCustomTypes = opts.unsafeAllowCustomTypes ?? false;
+  const unsafeAllowLinkedLibraries = opts.unsafeAllowLinkedLibraries ?? false;
+  const unsafeAllow = opts.unsafeAllow ?? [];
```

Comment:
I think this function should always return options that are consistent: `unsafeAllowCustomTypes` should be `true` if and only if `unsafeAllow.includes('struct-definition')` and `unsafeAllow.includes('enum-definition')`. Right now it's only ensuring a single direction (left to right).

An easy way to ensure the other direction is to set the initial values based on the array:
```suggestion
  const unsafeAllow = opts.unsafeAllow ?? [];
  const unsafeAllowCustomTypes =
    opts.unsafeAllowCustomTypes ||
    (unsafeAllow.includes('struct-definition') &&
      unsafeAllow.includes('enum-definition'));
  const unsafeAllowLinkedLibraries =
    opts.unsafeAllowLinkedLibraries ||
    unsafeAllow.includes('external-library-linking');
```

Just a suggestion, feel free to implement in a different way.